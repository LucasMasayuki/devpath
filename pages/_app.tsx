import { AppProps } from 'next/app'
import React from 'react'
import ThemeContainer from '@/contexts/theme/ThemeContainer'
import '@/styles/css/markdown-styles.css'

export default function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ThemeContainer>
      <Component {...pageProps} />
    </ThemeContainer>
  )
}
