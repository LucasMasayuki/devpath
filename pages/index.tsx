import Head from 'next/head'
import React from 'react'

import PostList from '@/components/post-list'
import AppBar from '@/components/app-bar'
import Layout from '@/components/layout'
import { getAllPosts } from '@/lib/api'
import Post from '@/types/post'

type Props = {
  allPosts: Post[]
}

const Index = ({ allPosts }: Props) => {
  const posts = allPosts

  return (
    <>
      <Layout>
        <Head>
          <title>
              Devpath Blog
          </title>
        </Head>
        <AppBar />
  
        {posts.length > 0 && <PostList posts={posts} />}
      </Layout>
    </>
  )
}

export default Index

export const getStaticProps = async () => {
  const allPosts = getAllPosts([
    'title',
    'date',
    'slug',
    'author',
    'coverImage',
    'excerpt',
  ])

  return {
    props: { allPosts },
  }
}
