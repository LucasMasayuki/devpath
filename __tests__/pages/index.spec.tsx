import React from 'react'
import renderer from 'react-test-renderer'
import Index from '@/pages/index'
import { getStaticProps } from '@/pages/index'
import Author from '@/types/author'
import Post from '@/types/post'

describe('Index', () => {
  it('should render Index', () => {
    const author: Author = {
      name: 'test',
      picture: 'test',
    }

    const post: Post = {
      content: 'test',
      slug: '',
      title: 'test',
      coverImage: 'test.com',
      date: '1996-10-02',
      author,
      excerpt: 'test',
      ogImage: {
        url: 'test',
      },
    }

    const posts: Post[] = [post]

    const component = renderer.create(<Index allPosts={posts} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should get static props', async () => {
    const staticProps = await getStaticProps()
    expect(staticProps.props.allPosts.push.length).not.toEqual(0)
  })
})
