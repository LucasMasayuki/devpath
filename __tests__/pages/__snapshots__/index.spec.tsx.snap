// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render Index 1`] = `
Array [
  <main>
    <div
      className="css-173ple2"
    >
      <div
        className="css-k008qs"
      >
        <div
          className="css-iufztw"
        >
          <h2
            className="chakra-heading css-in3yi3"
          >
            <a
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              DEVPATH
            </a>
          </h2>
        </div>
        <div
          className="css-17xejub"
        />
        <div
          className="css-0"
        >
          <button
            aria-label="Menu"
            className="chakra-button css-1kgd4fd"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-1aplkls"
    >
      <div
        className="css-0"
      >
        <div
          className="css-dv71fa"
        >
          <div
            className="css-qy636i"
          >
            <div
              style={
                Object {
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "left": 0,
                  "margin": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <img
                alt="Cover Image for test"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "visibility": "hidden",
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
          <div
            className="css-n16x53"
          >
            <div
              className="css-j7qwjs"
            >
              <div
                className="css-0"
              >
                <h3
                  className="chakra-heading css-1dv4iqg"
                >
                  <a
                    className="chakra-link css-0"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    test
                  </a>
                </h3>
              </div>
              <div
                className="css-1yh2cij"
              >
                <span
                  className="chakra-avatar css-1ruhx4w"
                >
                  <div
                    aria-label="test"
                    className="chakra-avatar__initials css-0"
                  >
                    t
                  </div>
                </span>
                <span
                  className="css-xf6agq"
                >
                  test
                </span>
              </div>
              <div
                className="css-eg4v4b"
              >
                <time
                  dateTime="1996-10-02"
                >
                  October	2, 1996
                </time>
                 - 
                <svg
                  className="chakra-icon css-1kmqiqp"
                  focusable={false}
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M23.432,10.524C20.787,7.614,16.4,4.538,12,4.6,7.6,4.537,3.213,7.615.568,10.524a2.211,2.211,0,0,0,0,2.948C3.182,16.351,7.507,19.4,11.839,19.4h.308c4.347,0,8.671-3.049,11.288-5.929A2.21,2.21,0,0,0,23.432,10.524ZM7.4,12A4.6,4.6,0,1,1,12,16.6,4.6,4.6,0,0,1,7.4,12Z"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="2"
                    />
                  </g>
                </svg>
                <small>
                  0 views
                </small>
              </div>
              <div
                className="css-1fhgjcy"
              >
                <p
                  className="chakra-text css-1aayjs6"
                >
                  test
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <footer>
    <div
      className="css-14i12oi"
    >
      <div
        className="css-150970n"
      >
        <h2
          className="chakra-heading css-in3yi3"
        >
          DEVPATH
        </h2>
        <div
          className="css-1l2zrv3"
        >
          <div
            className="css-1m1l6q9"
          >
            <p
              className="chakra-text css-0"
            >
              Created by Lucas Masayuki
            </p>
          </div>
          <div
            className="css-4xeohy"
          >
            <a
              className="chakra-link css-0"
              href="https://github.com/LucasMasayuki"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                aria-label="My personal github"
                className="chakra-button css-1kgd4fd"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="chakra-icon css-13otjrl"
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </button>
            </a>
            <a
              className="chakra-link css-0"
              href="https://br.linkedin.com/in/lucas-tamaribuchi"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                aria-label="My Linkedin"
                className="chakra-button css-1kgd4fd"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="chakra-icon css-13otjrl"
                  fill="currentColor"
                  focusable={false}
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  />
                </svg>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>,
]
`;
